app/bundles/LeadBundle/Entity/LeadListRepository.php

```diff
@@ -864,4 +864,27 @@ public function getLeadSegmentIds(int $leadId): array
 
         return array_column($result, 'id');
     }
+
+    /**
+     * Get segment IDs for a contact.
+     *
+     * @param string $contactId Contact ID (supports BIGINT UNSIGNED)
+     *
+     * @return int[]
+     */
+    public function getContactSegmentIds(string $contactId): array
+    {
+        return array_map('intval', $this->getEntityManager()
+            ->getConnection()
+            ->createQueryBuilder()
+            ->select('ll.leadlist_id')
+            ->from(MAUTIC_TABLE_PREFIX.'lead_lists_leads', 'll')
+            ->innerJoin('ll', MAUTIC_TABLE_PREFIX.'lead_lists', 'l', 'll.leadlist_id = l.id')
+            ->where('ll.lead_id = :contactId')
+            ->andWhere('ll.manually_removed = 0')
+            ->andWhere('l.is_published = 1')
+            ->setParameter('contactId', $contactId)
+            ->executeQuery()
+            ->fetchFirstColumn());
```

Comment:
Isn't there a simpler way how to return int[]? Something like
```suggestion
        return $this->getEntityManager()
            ->getConnection()
            ->createQueryBuilder()
            ->select('ll.leadlist_id')
            ->from(MAUTIC_TABLE_PREFIX.'lead_lists_leads', 'll')
            ->innerJoin('ll', MAUTIC_TABLE_PREFIX.'lead_lists', 'l', 'll.leadlist_id = l.id')
            ->where('ll.lead_id = :contactId')
            ->andWhere('ll.manually_removed = 0')
            ->andWhere('l.is_published = 1')
            ->setParameter('contactId', $contactId)
            ->executeQuery()
            ->fetchAllNumeric();
```
I don't like using method names as strings. There is an eval function in the implementation somewhere and that is terrible. An arrow function would be better if Doctrine cannot produce int[] by itself.